radixdlt/babylon-wallet-android
# Manual account recovery

## Description
Flow is accessible in two places:
- backup recovery -> other restore options
- settings -> account security -> Account Recovery scan
Here is brief description of flows:
- Onboarding screen: this flow allow to add manually BDFS which will be marked as main. Since we have "empty wallet" we scan starting from 0 index in batches of 50.
- settings: We can scan using existing mnemonics (both olympia and babylon) as well as ledgers. Indices to scan are computed so that we always start from 0 and move incrementally, excluding existing account indices. Babylon seed phrases added here are not MAIN. Add ledger flow is shared with the one that is used in creating new account.

Changes:
- fixed problem with backup recovery when we skip main bdfs and enter 2nd seed phrase - we discussed it at length.
- same seed phrase can be entered as babylon and as Olympia. If this happen, we need to list one seed phrase as two entries in Seed phrases screen and in seed phrase selection
- changed how we compute derivation indices. It is computed per network, per factor source and per curve used (since we can have same mnemonic in olympia/babylon - thus we differentiate it by curve). We find max account index by parsing actual account derivation path, and we pick next int.
- I also found out that we assign wrong derivation scheme to Olympia accounts - CAP_26 instead of BIP44. I've added code that runs on app startup and fixes that. To test this scenario, please import some olympia accounts using our main branch, and inspect profile in settings, what is derivation path scheme for olympia accounts. After installing this branch, inspect profile again to see that this was fixed.
- accounts added in recovery scan have their default deposit rule synced with gateway, and specific assets/depositor rules are now set as NULL instead of empty array. This is then reflected in 3rd party deposit settings, they have unknown status and warning message.
- olympia recovery options should not be visible on mainnet.
- recovery scan used from onboarding uses SHARED view model, so seed phrase screen sets "virtual" factor source to that view model, and profile with new recovered accounts is saved at the very end. Settings flows don't use shared view model becuase factor sources are already saved, so passing their id is enough. The reason why I didn't store onboarding virtual factor source is that we would need to store mnemonic too, so just storing it and reading on next screen would require 2 additional biometric prompts every time you advance from seed phrase input to recovery scan.

## How to test

Here is seed phrase you can use in recovery, that can be used both as olympia and babylon:
`wine over village stage barrel strategy cushion decline echo fiber salad carry empower fun awful cereal galaxy laundry practice appear bean flat mansion license`.
You can also use any other seed phrase that you control and know that it has active accounts. Active account is one used in at least 1 transaction.

## Screenshot

Example:

## Video

## PR submission checklist
